sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ParameterValidationInAsyncShouldBeWrapped.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
```

Comment:
Usual annoying comment: please add tests for local functions.

Reply:
I have added a test but I am not sure what should be the behavior.

Reply:
That seems like a nice pattern if outer method is sync (and validates args) and inner function is async - does exactly what we want and to the outside world still looks like one method.

To support that, we'd need to check if async local function does any argument checking and is inside sync method. Up to you if you want to handle that (could be in a separate ticket).

At minimum, let's document (in test) the behaviour of that case. Pseudo-code:
```
public void Foo(int x)
{
    async localFoo(int y)
    {
        if (y == 0) throw new ArgumentNullException("y");
    }
}
```